lib/AST/ASTContext.cpp

```diff
@@ -563,6 +565,14 @@ void ASTContext::setLazyResolver(LazyResolver *resolver) {
   }
 }
 
+void ASTContext::addLazyParser(LazyMemberParser *lazyParser) {
+  getImpl().lazyParsers.insert(lazyParser);
```

Comment:
Why can there be more than one lazy parser?

Reply:
Currently, lazy parser is implemented via `PersistentParserState`, which is a per-file entity. So we have multiple parsers. I'll see how can we refactor them into one.